src/coreclr/src/inc/eventtracebase.h

```diff
@@ -340,37 +352,54 @@ class XplatEventLoggerConfiguration
         {
             auto const length = component.End - component.Start;
             providerName = new WCHAR[length + 1];
-            memset(providerName, '\0', (length + 1) * sizeof(WCHAR));
             wcsncpy(providerName, component.Start, length);
+            providerName[length] = '\0';
         }
         return providerName;
     }
 
-    ULONGLONG ParseEnabledKeywordsMask(ComponentSpan const & component) const
+    uint64_t ParseEnabledKeywordsMask(ComponentSpan const & component) const
     {
-        auto enabledKeywordsMask = (ULONGLONG)(-1);
+        auto enabledKeywordsMask = (uint64_t)(-1);
         if ((component.End - component.Start) != 0)
         {
             enabledKeywordsMask = _wcstoui64(component.Start, nullptr, 16);
         }
         return enabledKeywordsMask;
     }
 
-    UINT ParseLevel(ComponentSpan const & component) const
+    uint32_t ParseLevel(ComponentSpan const & component) const
     {
-        auto level = TRACE_LEVEL_VERBOSE;
+        int level = TRACE_LEVEL_VERBOSE; // Verbose
         if ((component.End - component.Start) != 0)
         {
             level = _wtoi(component.Start);
         }
         return level;
     }
 
+    LPCWSTR ParseArgument(ComponentSpan const & component) const
+    {
+        auto argument = (WCHAR*)nullptr;
+        if ((component.End - component.Start) != 0)
+        {
+            auto const length = component.End - component.Start;
+            argument = new WCHAR[length + 1];
```

Comment:
This allocation appears to leak. I think the pre-existing allocation for provider name is also leaking. Using NewHolder on the _argument and _provider fields should fix it.